inductiva/resources/machines.py

```diff
@@ -0,0 +1,112 @@
+"""Containes MachineGroup class to manage GC resources."""
+from absl import logging
+from typing import Optional
+import uuid
+import inductiva
+from inductiva import api
+from inductiva import resources
+
+from inductiva.client import ApiClient, ApiException
+from inductiva.client.apis.tags.instance_api import InstanceApi
+from inductiva.client.model.instance import Instance
+from inductiva.client.model.instance_group import InstanceGroup
+
+
+class MachineGroup():
+    """Class to manage Google Cloud resources."""
+
+    def __init__(
+        self,
+        machine_type: str,
+        num_machines: int = 1,
+        spot: bool = False,
+        disk_size_gb: int = 30,
+        label: str = None,
+        machine_group_id: Optional[uuid.UUID] = None,
+    ) -> None:
+        """Create a MachineGroup object.
+
+        Args:
+            machine_type: The type of GC machine to launch. Ex: "e2-medium".
+            num_machines: The number of machines to launch.
+            spot: Whether to use spot instances.
+            disk_size_gb: The size of the disk in GB. (min. 30 GB)
+            label: The label to assign to the machine group.
+            machine_group_id: If None the Machine Group ID will be generated
+              automatically. A new ID can be created by invoking the
+              inductiva.resources.create_resource_pool() method.
+        """
+        self.machine_type = machine_type
+        self.num_machines = num_machines
+        self.spot = spot
+        self.disk_size_gb = disk_size_gb
+        self.label = label
```

Comment:
why label if there's already a name? The user can just simply optionally pass the name or not, do we need the label?

Reply:
Label is just for the user her/himself to make a note of the machine group, it is not a name. Passing the name of the machine group is not a good idea because that's what is used to create resources and we need to ensure that each resource has a unique name.